chia/full_node/mempool_check_conditions.py

```diff
@@ -132,40 +134,219 @@ def mempool_assert_my_amount(condition: ConditionWithArgs, unspent: CoinRecord)
+    """
+    op = ConditionOpcode
+    cc = ConditionCost
+    if condition is op.AGG_SIG_UNSAFE or condition is op.AGG_SIG_ME:
+        return cc.AGG_SIG.value, parse_aggsig(args)
+    elif condition is op.CREATE_COIN:
+        return cc.CREATE_COIN.value, parse_create_coin(args)
+    elif condition is op.ASSERT_SECONDS_ABSOLUTE:
+        return cc.ASSERT_SECONDS_ABSOLUTE.value, parse_seconds(args, Err.ASSERT_SECONDS_ABSOLUTE_FAILED)
+    elif condition is op.ASSERT_SECONDS_RELATIVE:
+        return cc.ASSERT_SECONDS_RELATIVE.value, parse_seconds(args, Err.ASSERT_SECONDS_RELATIVE_FAILED)
+    elif condition is op.ASSERT_HEIGHT_ABSOLUTE:
+        return cc.ASSERT_HEIGHT_ABSOLUTE.value, parse_height(args, Err.ASSERT_HEIGHT_ABSOLUTE_FAILED)
+    elif condition is op.ASSERT_HEIGHT_RELATIVE:
+        return cc.ASSERT_HEIGHT_RELATIVE.value, parse_height(args, Err.ASSERT_HEIGHT_RELATIVE_FAILED)
+    elif condition is op.ASSERT_MY_COIN_ID:
+        return cc.ASSERT_MY_COIN_ID.value, parse_coin_id(args, Err.ASSERT_MY_COIN_ID_FAILED)
+    elif condition is op.RESERVE_FEE:
+        return cc.RESERVE_FEE.value, parse_fee(args)
+    elif condition is op.CREATE_COIN_ANNOUNCEMENT:
+        return cc.CREATE_COIN_ANNOUNCEMENT.value, parse_announcement(args)
+    elif condition is op.ASSERT_COIN_ANNOUNCEMENT:
+        return cc.ASSERT_COIN_ANNOUNCEMENT.value, parse_hash(args, Err.ASSERT_ANNOUNCE_CONSUMED_FAILED)
+    elif condition is op.CREATE_PUZZLE_ANNOUNCEMENT:
+        return cc.CREATE_PUZZLE_ANNOUNCEMENT.value, parse_announcement(args)
+    elif condition is op.ASSERT_PUZZLE_ANNOUNCEMENT:
+        return cc.ASSERT_PUZZLE_ANNOUNCEMENT.value, parse_hash(args, Err.ASSERT_ANNOUNCE_CONSUMED_FAILED)
+    elif condition is op.ASSERT_MY_PARENT_ID:
+        return cc.ASSERT_MY_PARENT_ID.value, parse_coin_id(args, Err.ASSERT_MY_PARENT_ID_FAILED)
+    elif condition is op.ASSERT_MY_PUZZLEHASH:
+        return cc.ASSERT_MY_PUZZLEHASH.value, parse_hash(args, Err.ASSERT_MY_PUZZLEHASH_FAILED)
+    elif condition is op.ASSERT_MY_AMOUNT:
+        return cc.ASSERT_MY_AMOUNT.value, parse_amount(args)
+    else:
+        raise ValidationError(Err.INVALID_CONDITION)
+
+
+CONDITION_OPCODES: Set[bytes] = set(item.value for item in ConditionOpcode)
+
+
+def parse_condition(cond: SExp, safe_mode: bool) -> Tuple[int, Optional[ConditionWithArgs]]:
+    condition = cond.first().as_atom()
+    if condition in CONDITION_OPCODES:
+        opcode: ConditionOpcode = ConditionOpcode(condition)
+        cost, args = parse_condition_args(cond.rest(), opcode)
+        cvl = ConditionWithArgs(opcode, args) if args is not None else None
+    elif not safe_mode:
+        opcode = ConditionOpcode.UNKNOWN
+        cvl = ConditionWithArgs(opcode, cond.rest().as_atom_list())
+        cost = 0
+    else:
+        raise ValidationError(Err.INVALID_CONDITION)
+    return cost, cvl
+
+
+def get_name_puzzle_conditions(
+    generator: BlockGenerator, max_cost: int, *, cost_per_byte: int, safe_mode: bool
```

Comment:
is the "*" a typo ?

Reply:
I added it to force all parameters on the right of it to be named at all call-sites. two `int`s and one `bool` can make it very hard to know what they mean at call sites otherwise. I made this worse by adding yet another `int` parameter, so I thought it was a good idea to force callers to name the arguments.

This is at least my understanding of what `*` does in python, but I just read that on the internet somewhere.

Reply:
cool, i didn't know about "*" syntax not sure how common it is, i would explain it in the description if its not something common